WeatherForecast/WeatherForecast/Model/WeatherListTableDelegate.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  WeatherListTableDelegate.swift
+//  WeatherForecast
+//
+//  Created by 구 민영 on 3/19/24.
+//
+
+import UIKit
+
+final class WeatherListTableDelegate: NSObject, UITableViewDelegate {
+    private let baseVC: UIViewController
+    var weathers: [WeatherForecastInfo]?
+    var city: City?
+    
+    convenience init(baseVC: UIViewController) {
+        self.init(baseVC: baseVC, weathers: nil, city: nil)
+    }
+    
+    init(baseVC: UIViewController, weathers: [WeatherForecastInfo]?, city: City?) {
+        self.baseVC = baseVC
+        self.weathers = weathers
+        self.city = city
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        guard let weathers,
+              let city else {return}
```

Comment:
Swift는 일반적으로 중괄호 전후로 공백을 두는 컨벤션을 사용해요.

```suggestion
              let city else { return }
```